packages/amplify-category-auth/src/provider-utils/awscloudformation/assets/string-maps.js

```diff
@@ -423,19 +450,19 @@ const authorizeScopes = [
 const signInOptions = [
   {
     name: 'Username',
-    value: 'username',
+    value: ['username'],
   },
   {
     name: 'Email',
-    value: 'email',
+    value: ['email'],
   },
   {
     name: 'Phone Number',
-    value: 'phone_number',
+    value: ['phone_number'],
   },
   {
     name: 'Email or Phone Number',
-    value: 'email, phone_number',
+    value: ['email', 'phone_number'],
```

Comment:
because it was confusing to have to parse a string to figure out if it was an array and set stuff accordingly in the business logic

Reply:
We can not make this modification because there are projects that are set up by older versions of the CLI, and in those project, the value is "email, phone_number", the updated CLI will then throw error on those old projects. 
If we decide to make this update, then we have to write migration logic on those projects, which can be very messy and error prone.

Reply:
We can not make this modification because there are projects that are set up by older versions of the CLI, and in those project, the value is "email, phone_number", the updated CLI will then throw error on those old projects.
If we decide to make this update, then we have to add migration logic for those projects, which can be very messy and error prone.